config_test.go

```diff
@@ -80,6 +111,22 @@ func TestConfig(t *testing.T) {
 			for i := 0; i < 200; i++ {
 				logger.Info("sampling")
 			}
+
+			// Rolling over tick to produce sampling report.
+			if tt.expectSamplingN > 0 {
+				ztest.Sleep(time.Second)
```

Comment:
this is a long sleep which will slow down the test execution. In general, use of sleep is bad for tests (slow and also cause flakiness). I think we can't avoid it for the sampler tests currently, but the other tests are doing milliseconds rather than seconds to avoid a large impact to the test suite runtime.